polynote-frontend/polynote/cell.js

```diff
@@ -106,7 +106,16 @@ export class Cell extends UIEventTarget {
         this.cellInput.querySelector('.run-cell').onclick = (evt) => {
             this.dispatchEvent(new RunCellEvent(this.id));
         }
+    }
 
+    static extractId(cellRef) {
```

Comment:
Is this so you don't have to worry about whether you're using the container element or the cell object?

Reply:
Yep! Mainly because the Toolbar/MainUI and the NotebookUI don't really talk to each other right now, so the toolbar stuff doesn't have an easy way to reference the Cell objects - it has to go through the containers. 

It shouldn't be too hard to add such a reference so we could get rid of this query selection business though.

Reply:
The container has a reference to the cell object though